drivers/hud_spm_report/app/models/hud_spm_report/fy2024/episode.rb

```diff
@@ -286,6 +287,9 @@ def compute_episode(enrollments, included_project_types:, excluded_project_types
       bed_nights = existing_bed_nights.index_by(&:last)
       enrollments.each do |enrollment|
         if enrollment.project_type.in?(HudUtility2024.project_type_number_from_code(:es_nbn))
+          # FIXME, see #7473
```

Comment:
@ttoomey I think this is reasonable, though it feels like this shouldn't be happening.

If we called `references` and `preload` on the scope, would that avoid this situation by making sure the data is fetched when the query starts?

Reply:
I don't think that preload or references would be a robust solution. My thought is that we need synchronization between reporting and daily processing tasks, probably via an advisory lock

Reply:
Yeah, that wasn't a proposal as a full-featured solution, but as a way to decrease the possible data collisions.  

Advisory locks might work, but the import process can take hours, as can some report jobs, so we'd need to find the right granularity that would allow both to run without affecting the other too much.